src/core/Target_file.ml

```diff
@@ -0,0 +1,19 @@
+type provided_target_language_info =
+  (* The default *)
+  | NoSpecialTargeting
+  (* When `--scan-unknown-extensions` is provided, explicitly passed files
+     are scanned with all languages in the rule *)
+  | ExplicitFileScanUnknownExtension (* | TODO Lang for -lang? *)
+[@@deriving show]
+
+type target_file = Fpath.t * provided_target_language_info [@@deriving show]
+type target_files = target_file list [@@deriving show]
```

Comment:
remove target_files, Target_file.t list is as short as Target_file.target_files

Reply:
well, the declaration can be useful as it provides `show_target_files` but I wouldn't use the type name and would favor `Target_file.t list`.